tests/unit/samplegen/test_samplegen.py

```diff
@@ -1048,6 +1317,175 @@ def test_validate_expression_base_attr_is_repeated():
     v.validate_expression("molluscs[0].mantle")
 
 
+def test_validate_expression_map_lookup():
+    # See https://github.com/protocolbuffers/protobuf/blob/master/src/google/protobuf/descriptor.proto#L475
+    # for details on how mapped attributes get transformed by the protoc compiler.
+    OutputType = DummyMessage(
+        fields={
+            "cephalopods": DummyField(
+                message=DummyMessage(
+                    fields={
+                        # real type is most likely str in real protos
+                        "key": DummyField(),
+                        "value": DummyField(
+                            message=DummyMessage(
+                                fields={
+                                    "mantle": DummyField(
+                                        message=DummyMessage(type="MANTLE_TYPE",
+                                                             fields={}),
+                                    )
+                                },
+                                type="CEPHALOPOD_TYPE"
+                            )
+                        ),
+                    },
+                    type="CEPHALOPODS_TYPE",
+                    options=namedtuple("MessageOptions", ["map_field"])(True)),
+                repeated=True,
+            )
+        },
+        type="MOLLUSC_TYPE"
+    )
+    method = DummyMethod(output=OutputType)
+    v = samplegen.Validator(method)
+    v.validate_expression('$resp.cephalopods{"squid"}.mantle')
+
+
+def test_validate_expression_map_lookup_terminal_lookup():
```

Comment:
Which test case checks the non-indexed terminal collection passing case and the non-indexed, non-terminal collection failing test case?

Reply:
`test_validate_expression_repeated_attrs` verifies the non-indexed, non-terminal collection failure. I'll rename the testcase to make that more clear.

Reply:
And non-indexed terminal collection?

Reply:
A non-indexed terminal collection is explicitly allowed.
```
{"define": "molluscs=$resp.molluscs"}
{"loop": {"collection": "molluscs", "variable": "m", "body": [{"print": ["Mollusc: %s", "m"]}]}}
```

Reply:
Right, and you have a test for that, correct?

Reply:
Yes, `test_validate_expression_base_attr_is_repeated`